src/Discovery/Discovery.tsx

```diff
@@ -0,0 +1,552 @@
+                  <span className='matched'>{value.slice(matchIndex, matchIndex + searchTerm.length)}</span>
+                  {value.slice(matchIndex + searchTerm.length)}
+                </React.Fragment>
+                );
+              }
+              return value;
+            };
+            return (
+              <div
+                className='discovery-table__expanded-row-content'
+                role='button'
+                tabIndex={0}
+                onClick={() => {
+                  setModalData(record);
+                  setModalVisible(true);
+                }}
+              >
+                {renderValue(studyPreviewText)}
+              </div>
+            );
+          },
+          expandedRowClassName: () => 'discovery-table__expanded-row',
+          expandIconColumnIndex: -1, // don't render expand icon
+        })}
+      />
+    </div>
+    <Modal
+      className='discovery-modal'
+      visible={modalVisible}
+      onOk={() => setModalVisible(false)}
+      onCancel={() => setModalVisible(false)}
+      width='80vw'
+      footer={false}
+    >
+      <Space direction='vertical' size='large'>
+        { config.study_page_fields.header &&
+          <Space align='baseline'>
+            <h3 className='discovery-modal__header-text'>{modalData[config.study_page_fields.header.field]}</h3>
+            <a href={`/discovery/${modalData[config.minimal_field_mapping.uid]}/`}><LinkOutlined /> Permalink</a>
+          </Space>
+        }
+        { config.features.authorization.enabled &&
+          (modalData[accessibleFieldName]
+            ? (
+              <Alert
+                className='discovery-modal__access-alert'
+                type='success'
+                message={<><UnlockOutlined /> You have access to this study.</>}
+              />
+            )
+            : (
+              <Alert
+                className='discovery-modal__access-alert'
+                type='warning'
+                message={<><LockFilled /> You do not have access to this study.</>}
+              />
+            )
+          )
+        }
+        { config.study_page_fields.fields_to_show.map((fieldGroup, i) => (
+          <div key={i} className='discovery-modal__attribute-group'>
+            { fieldGroup.include_name &&
+                  <h3 className='discovery-modal__attribute-group-name'>{fieldGroup.group_name}</h3>
+            }
+            { fieldGroup.fields.map((field) => {
+              // display nothing if selected study doesn't have this field
+              // and this field isn't configured to show a default value
+              if (!modalData[field.field] && !field.include_if_not_available) {
+                return null;
+              }
+              return (
+                <div key={field.name} className='discovery-modal__attribute'>
+                  { field.include_name !== false &&
+                        <span className='discovery-modal__attribute-name'>{field.name}</span>
```

Comment:
minor: personally i'd include `:` after `field.name` so iot will show as `{field.name}: {value}`

Reply:
This is a good idea for readability, I'm adding it.